soft4pes/utils/sequence.py

```diff
@@ -59,11 +46,19 @@ def __call__(self, t):
 
         Returns
         -------
-        1 x 2 ndarray of floats
+        1 x m ndarray of floats
             Interpolated output.
 
         """
-        return np.array([
-            np.interp(t, self.times, self.values[:, 0], period=self._period),
-            np.interp(t, self.times, self.values[:, 1], period=self._period)
-        ])
+        # Define the dimensions of columns to iterate over
+        dim = range(self.values.shape[1])
+
+        #Initialize a list to hold the interpolated values
+        interpolated_values = []
+
+        # Perform interpolation for each dimension
```

Comment:
For each column. I may have used word dimension in the task description, but the correct term is column (we always have two dimensions)